### ISSUE TITLE
get_initial_state beautification

### ISSUE THREAD

**Issue by flomlo** [fchollet/keras#7808]
So, in recurrent.py is a piece of batshit-crazy code:

    def get_initial_state(self, inputs):
        # build an all-zero tensor of shape (samples, output_dim)
        initial_state = K.zeros_like(inputs)  # (samples, timesteps, input_dim)
        initial_state = K.sum(initial_state, axis=(1, 2))  # (samples,)
        initial_state = K.expand_dims(initial_state)  # (samples, 1)
        initial_state = K.tile(initial_state, [1, self.units])  # (samples, output_dim)
        initial_state = [initial_state for _ in range(len(self.states))]
    return initial_state

I do understand that K.shape is no solution for this mess as it needs to evaluate a graph at build time (not nice), but wouldnt K.int_shape solve the problem quite nicely? Kinda like that:

    def get_initial_state(self, inputs):
        # build an all-zero tensor of shape (samples, output_dim)
        samples = K.int_shape(inputs)[0]
        initial_state = K.zeros(shape=(samples, self.units))  # (samples, output_dim)
    return initial_state    

Or is there an issue I'm overlooking?
Otherwise im tempted to cook up a patch in a minute.

-------------
- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps

- [x] If running on TensorFlow, check that you are up-to-date with the latest version. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

**Comment by samre12** [comment]
@flomlo just a small change, you need to add another line to your code 
```python
initial_state = [initial_state for _ in range(len(self.states))]
```

**Comment by flomlo** [comment]
Ok, I tested it, and it fails (in hindsight: of course) for the very common case that K.int_shape(inputs) is e.g. (None, None, 5).

Therefore: Stupid idea. Closing Thread.
